homeassistant/components/number/__init__.py

```diff
@@ -48,12 +48,22 @@ async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
     component.async_register_entity_service(
         SERVICE_SET_VALUE,
         {vol.Required(ATTR_VALUE): vol.Coerce(float)},
-        "async_set_value",
+        async_set_value,
     )
 
     return True
 
 
+async def async_set_value(entry: NumberEntity, async_service_call: Callable) -> None:
+    """Service call wrapper to set a new value."""
+    value = getattr(async_service_call, "data")["value"]
```

Comment:
```suggestion
    value = service_call.data["value"]
```

Reply:
We were missing the ServiceCall import. Added that.